### ISSUE TITLE
Broken in Deno 2.0

### ISSUE THREAD

**Issue by gibbygano** [xstevenyung/fresh-seo#27]
After upgrading your environment to Deno 2.0 project is unable to build, showing error: 

```
error: Uncaught (in promise) ReferenceError: window is not defined
if (!window.hackle) window.hackle = makeHackle()
^
    at https://deno.land/x/hackle@1.1.1/init.ts:9:1

    info: window global is not available in Deno 2.
    hint: Replace `window` with `globalThis`.
```

Pretty clear from the error message, hackle uses window which is no longer valid in Deno 2.0. 
Hackle itself hasn't been updated since January of 2021: [Hackle on Github](https://github.com/Vehmloewff/hackle).
Not sure if I'd hold my breathe for an update from them.

**Comment by gibbygano** [comment]
I went ahead and left an issue on the hackle repository just in case someone is monitoring the issues there still.

**Comment by m4rocks** [comment]
hi @gibbygano! this issue has nothing to do with fresh_seo. we don't even use hackle :)

**Comment by gibbygano** [comment]
@notangelmario When I search the repo hackle comes up and when I removed fresh-seo the error was resolved and the project ran. 

![image](https://github.com/user-attachments/assets/9b45ae4b-8d79-45c5-8e13-6f0622b78bf9)

**Comment by m4rocks** [comment]
huh. very weird. i'll look into it. reopening...

**Comment by Hexagon** [comment]
+1, unable to run a site on Deno 2 because of this hackle dependency

**Comment by scsmash3r** [comment]
Run into this issue too.

`fresh-seo` package has a dependency, called `glob-filter` - it is used in `remove` function of sitemap entries.
https://github.com/xstevenyung/fresh-seo/blob/71236baf0186bedfe39b76ab5aef862f1e1ae3d0/src/deps.ts#L14

The `glob-filter` package has this `hackle` dependency at the very first line:
https://github.com/Vehmloewff/glob-filter/blob/4fe7a7d2a9e185126f14236a47a4f2015d626f63/mod.ts#L1

Since `glob-filter` package is not maintained and seems to be stalled, one could reimplement `filterFiles` function without `hackle` dependency.

This is probably what is called "a dependency hell", and with even more repos like crux land, deno land, and now - jsr... It is even more hardcore, when some feature breaks in major updates of a runtime.

**Comment by scsmash3r** [comment]
Quick dirty fix can be found here: https://github.com/scsmash3r/fresh-seo/commit/773f5948f16ad12e7f543b13bb263470aa431c3d

It is kinda working, but I have some issues with `manifest` taken from `fresh.gen.ts` file on a first run (when file is not created yet), so I think there must be some sort of a hook, where `fresh-seo` must init only after `fresh` generated it's manifest file. Couple of other issues too, but it can be worked out I think.

**Comment by scsmash3r** [comment]
This one works for Deno 2.0: https://github.com/scsmash3r/fresh-seo/tree/main
Added a PR, would be nice if the plugin could be merged into this repo so not to lose users.

**Comment by Vehmloewff** [comment]
I've [fixed the upstream glob_filter](https://github.com/Vehmloewff/glob-filter/commit/d09c2fb932724d24937f1e2b86a38873cbcbe69c) dependency. Upgrading to `glob_filter@1.0.1` will fix this (see #30).
